temporian/implementation/numpy/operators/assign.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implementation for the Assign operator."""
+from typing import Dict, List
+
+import numpy as np
+from temporian.implementation.numpy.data.event import NumpyEvent
+from temporian.implementation.numpy.data.event import NumpyFeature
+
+
+def _get_equal_sampling_indexes(
+    sampling_1: "NumpySampling", sampling_2: "NumpySampling", index: tuple
+) -> np.ndarray:
+    """Get indexes of timestamps that are equal to sampling_1 & sampling_2
+      in a specific sampling index.
+
+    Args:
+        sampling_1 (NumpySampling): sampling of assignee event.
+        sampling_2 (NumpySampling): sampling of assigned event.
+        index (tuple): index to check.
+
+    Returns:
+        np.ndarray: indexes of timestamps that are equal to sampling_1 & sampling_2.
+    """
+    return np.where(sampling_1.data[index] == sampling_2.data[index])[0]
```

Comment:
We could move this utility function to below the main class def

Reply:
(or remove it and just drop an in-line comment, since it's a one liner and used in a single place)